packages/datadog-plugin-tedious/src/index.js

```diff
@@ -0,0 +1,133 @@
+    return function makeRequestWithTrace (request, packetType) {
+      const connectionConfig = this.config
+      const scope = tracer.scope()
+      const childOf = scope.active()
+      const query = getQuery(request)
+
+      if (!query) {
+        return scope.activate(childOf, () => makeRequest.apply(this, arguments))
+      }
+
+      const span = tracer.startSpan(`tedious.request`, {
+        childOf,
+        tags: {
+          [Tags.SPAN_KIND]: Kinds.CLIENT,
+          'db.type': 'mssql',
+          'span.type': 'sql',
+          'component': 'tedious',
+          'service.name': config.service || `${tracer._service}-mssql`,
+          'resource.name': query
+        }
+      })
+
+      addConnectionTags(span, connectionConfig)
+      addDatabaseTags(span, connectionConfig)
+      addQueryTags(span, request)
+
+      analyticsSampler.sample(span, config.analytics)
+      request.callback = tx.wrap(span, request.callback)
+
+      return scope.bind(makeRequest, span).apply(this, arguments)
+    }
+  }
+}
+
+function createWrapGetRowStream (tracer) {
+  return function wrapGetRowStream (getRowStream) {
+    return function getRowStreamWithTrace () {
+      const scope = tracer.scope()
+
+      const rowToPacketTransform = getRowStream.apply(this, arguments)
+      return scope.bind(rowToPacketTransform)
+    }
+  }
+}
+
+function getQuery (request) {
+  if (request.parameters) {
+    if (request.parameters.length === 0) {
+      return request.sqlTextOrProcedure
+    } else {
+      const statement = request.parametersByName.statement || request.parametersByName.stmt
+      return statement.value
+    }
+  }
+}
+
+function addConnectionTags (span, connectionConfig) {
+  span.setTag('out.host', connectionConfig.server)
+  span.setTag('out.port', connectionConfig.options.port)
+}
+
+function addDatabaseTags (span, connectionConfig) {
+  span.setTag('db.user', connectionConfig.userName || connectionConfig.authentication.options.userName)
+  span.setTag('db.name', connectionConfig.options.database)
+  span.setTag('db.instance', connectionConfig.options.instanceName)
+}
+
+function addQueryTags (span, request) {
+  const transformedPacketType = request.sqlTextOrProcedure
+  const resourceType = sqlMapping[transformedPacketType] || 'query'
+  span.setTag('resource.type', resourceType)
+}
+
+module.exports = [
+  {
+    name: 'tedious',
+    versions: [ '>=1.0.0' ],
+    patch (tedious, tracer, config) {
+      this.wrap(tedious, 'Request', createWrapRequestClass(tracer))
+      this.wrap(tedious, 'Connection', createWrapConnectionClass(tracer))
+
+      this.wrap(tedious.Connection.prototype, 'makeRequest', createWrapMakeRequest(tracer, config))
+
+      if (tedious.BulkLoad && tedious.BulkLoad.prototype.getRowStream) {
+        this.wrap(tedious.BulkLoad.prototype, 'getRowStream', createWrapGetRowStream(tracer))
+      }
+    },
+    unpatch (tedious) {
+      this.unwrap(tedious.Connection.prototype, 'makeRequest')
```

Comment:
Everything in `patch` should be unpatched.

Reply:
Good catch!

Reply:
Why are the constructors not unpatched?

Reply:
Ah, I'm gonna have to save references to the old constructors. Another good catch!

Reply:
Another good catch!